code/modules/religion/burdened/psyker.dm

```diff
@@ -258,36 +288,47 @@
 	spell_max_level = 1
 	invocation_type = INVOCATION_NONE
 	spell_requirements = SPELL_REQUIRES_NO_ANTIMAGIC
-	channel_message = span_notice("You focus on your weapon...")
+	channel_message = span_notice("You focus on your trigger fingers...")
 	charge_overlay_icon = 'icons/effects/effects.dmi'
 	charge_overlay_state = "purplesparkles"
-	channel_flags = IGNORE_HELD_ITEM
+	channel_time = 5 SECONDS
+	channel_flags = IGNORE_USER_LOC_CHANGE
 	var/boosted = FALSE
 	var/effect_time = 10 SECONDS
 
-/datum/action/cooldown/spell/charged/projectile_booster/Destroy()
+/datum/action/cooldown/spell/charged/psychic_booster/Destroy()
 	if(boosted)
 		stop_effects()
 	return ..()
 
-/datum/action/cooldown/spell/charged/projectile_booster/Remove(mob/living/remove_from)
+/datum/action/cooldown/spell/charged/psychic_booster/Remove(mob/living/remove_from)
 	if(boosted)
 		stop_effects()
 	return ..()
 
-/datum/action/cooldown/spell/charged/projectile_booster/cast(atom/cast_on)
+/datum/action/cooldown/spell/charged/psychic_booster/cast(atom/cast_on)
 	. = ..()
+	if(boosted)
+		return
 	ADD_TRAIT(cast_on, TRAIT_DOUBLE_TAP, type)
 	RegisterSignal(cast_on, COMSIG_PROJECTILE_FIRER_BEFORE_FIRE, PROC_REF(modify_projectile))
 	addtimer(CALLBACK(src, PROC_REF(stop_effects)), effect_time)
 
-/datum/action/cooldown/spell/charged/projectile_booster/proc/stop_effects()
+/datum/action/cooldown/spell/charged/psychic_booster/proc/stop_effects()
 	REMOVE_TRAIT(owner, TRAIT_DOUBLE_TAP, type)
 	UnregisterSignal(owner, COMSIG_PROJECTILE_FIRER_BEFORE_FIRE)
 
-/datum/action/cooldown/spell/charged/projectile_booster/proc/modify_projectile(datum/source, obj/projectile/bullet, atom/firer, atom/original_target)
-	bullet.homing_turn_speed = 60
-	bullet.set_homing_target(original_target)
+/datum/action/cooldown/spell/charged/psychic_booster/proc/modify_projectile(datum/source, obj/projectile/bullet, atom/firer, atom/original_target)
+	var/atom/target = original_target
+	if(isturf(target) || (isobj(target) && !target.density)) //if weird target, we try to compensate in our homing
+		for(var/mob/living/shooting_target in range(1, get_turf(target)))
+			target = shooting_target
+			break
```

Comment:
make sure this can't cause the bullets to choose the caster/firer as their new target